src/rfq/protocols/bebop/state.rs

```diff
@@ -25,8 +26,46 @@ impl ProtocolSim for BebopState {
         todo!()
     }
 
-    fn spot_price(&self, _base: &Token, _quote: &Token) -> Result<f64, SimulationError> {
-        todo!()
+    fn spot_price(&self, base: &Token, quote: &Token) -> Result<f64, SimulationError> {
+        let base_address = Address::from_slice(&base.address).to_string();
+        let quote_address = Address::from_slice(&quote.address).to_string();
+
+        // Since this method does not care about sell direction, we average the price of the best
+        // bid and ask
```

Comment:
I'm also not sure if this was the right way to interpret this. I assumed based on the docstring of the Trait and also the input variable naming. Let me know!

Reply:
yes that is good enouugh!